_posts/2016-04-25-watson-and-rails.md

```diff
@@ -0,0 +1,176 @@
+---
+layout: post
+title: "Connecting a Rails App to a Watson API"
+description: "How to Use IBM's Watson APIs in Ruby on Rails"
+category:
+tags: [Rails, APIs]
+author: olivia_reeve
+---
+{% include JB/setup %}
+
+## What is Watson?
+
+IBM's Watson supercomputer was specifically developed to answer questions on the game show _Jeopardy!_ and in 2011 competed against the best players in the show's history, eventually receiving the $1 million first place prize. You can learn more about Watson's history and it's future [here](http://www.techrepublic.com/article/ibm-watson-the-inside-story-of-how-the-jeopardy-winning-supercomputer-was-born-and-what-it-wants-to-do-next/).
+
+Watson uses natural language processing (NLP) and machine learning to extract information from large amounts of data. In it's run on _Jeopardy!_, Watson had access to four terabytes of disk storage consisting of 200 million pages of content, including the full text of [Wikipedia](https://en.wikipedia.org/wiki/Wikipedia).
+
+IBM has made a series of videos with various academics, athletes, and celebrities to showcase Watson's abilities, such as one featuring Carrie Fisher leading an [AI support group](https://www.youtube.com/watch?v=f8T8eWBmls0).
+
+![AI support group](https://img.youtube.com/vi/f8T8eWBmls0/0.jpg)
+
+## Watson for Developers
+
+Fortunately for developers, access to Watson's capabilities is at our fingertips. IBM's artificial intelligence service is available via numerous [APIs](http://www.ibm.com/smarterplanet/us/en/ibmwatson/developercloud/services-catalog.html), including both NLP and visual identification. We can use these APIs to extract data from text, such as tone, concept and personality insights, and language conversion. We can use Watson to convert text to speech (or vice versa), and recognize faces and objects from images.
+
+![API options]({{ BASE_PATH }}/assets/images/watson-and-rails/watson-apis.png)
+
+## Watson and Ruby
+
+For a Ruby on Rails project, I wanted a user of my web application to be able to upload a PDF and have its contents displayed on the page. Watson's [document conversion](http://www.ibm.com/smarterplanet/us/en/ibmwatson/developercloud/doc/document-conversion/) API looked like the perfect solution. I could input a Word, PDF, or HTML document and convert it to plain text, HTML, or JSON format, which I could then display on the page. Eureka!
+
+Unfortunately, my joy was short-lived. I was new to programming, so imagine my dismay when I discovered that this and other Watson APIs don't have any Ruby documentation. Not only that, but Ruby doesn't natively support uploading a file through an HTTP request.
+
+#### Here's how I made it work:
+
+My app was all about teachers being able to upload assignments so that students could view them and make submissions. Basically, it was for homework.
+
```

Comment:
Suggestion for wording: "My app allowed teachers to upload assignments and students to view, complete and make submissions - basically modernizing the homework process."